src/infra/dataverse/sparql/sparqlGateway.ts

```diff
@@ -22,11 +22,11 @@ export const sparqlGateway: DataversePort = {
     offset: number,
     filters: RetrieveDataverseQueryFilters
   ): TE.TaskEither<Error, RetrieveDataverseResult> => {
-    const buildContainsExpression = (filter: DataverseElementType): string =>
-      `contains(str(?type), "${filter}" )`
+    const buildStrExpression = (filter: DataverseElementType): string =>
+      `str(?type) = str(core:${filter})`
 
     const buildQueryFilter = (index: number, acc: string, cur: DataverseElementType): string =>
-      index === 0 ? buildContainsExpression(cur) : `${acc} || ${buildContainsExpression(cur)}`
+      index === 0 ? buildStrExpression(cur) : `${acc} || ${buildStrExpression(cur)}`
 
     const byTypeFilter = (f: DataverseElementType[]): string =>
```

Comment:
you can simplify by just considering that the filter is a join of eq expressions (provided by `buildStrExpression`):

```suggestion
       const byTypeFilter = (filters: DataverseElementType[]): string =>
        filters.map(buildStrExpression).join(' || ')
```

Reply:
Great, thank you for this improvement !

Reply:
I implemented it, but in `fp-ts` way 😜

Reply:
> I implemented it, but in `fp-ts` way 😜

far better, indeed ☺️